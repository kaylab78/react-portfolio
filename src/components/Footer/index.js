import React from 'react';

function Footer() {
    return (
        <footer>
            <ul>
                <li>
                    &copy; 2022 Kayla Backus
                </li>
                <li>
                    <a href="https://github.com/kaylab78" target="_blank" rel="noreferrer">GitHub</a>
                </li>
                <li>
                    <a href="https://www.linkedin.com/in/kaylabackus/" target="_blank" rel="noreferrer">LinkedIn</a>
                </li>
                <li>
                    <a href="https://stackoverflow.com/users/18535171/kayla" target="_blank" rel="noreferrer">Stack Overflow</a>
                </li>
            </ul>
        </footer>
    );
}

export default Footer;