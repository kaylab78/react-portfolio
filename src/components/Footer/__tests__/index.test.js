import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Footer from '..';

afterEach(cleanup);

describe('Footer component', () => {
    it('renders', () => {
        render(<Footer />);
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Footer />);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('links are visible', () => {
    it('inserts text into links', () => {
        const { getByTestId } = render(<Footer />);

        expect(getByTestId('github')).toHaveTextContent('GitHub');
        expect(getByTestId('linkedin')).toHaveTextContent('LinkedIn');
        expect(getByTestId('stackoverflow')).toHaveTextContent('Stack Overflow');
    });
});