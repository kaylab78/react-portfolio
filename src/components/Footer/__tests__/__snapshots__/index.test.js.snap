// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
<DocumentFragment>
  <footer>
    <ul>
      <li>
        <a
          href="https://github.com/kaylab78"
          rel="noreferrer"
          target="_blank"
        >
          GitHub
        </a>
      </li>
      <li>
        <a
          href="https://www.linkedin.com/in/kaylabackus/"
          rel="noreferrer"
          target="_blank"
        >
          LinkedIn
        </a>
      </li>
      <li>
        <a
          href="https://stackoverflow.com/users/18535171/kayla"
          rel="noreferrer"
          target="_blank"
        >
          Stack Overflow
        </a>
      </li>
    </ul>
  </footer>
</DocumentFragment>
`;
